### PR TITLE
countdown 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by antonmedv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/countdown.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+  end
+
+  test do
+    assert_equal "", pipe_output(bin/"countdown", "0m0s").strip
```
Comment (chenrui333) [inline Formula/countdown.rb:15]: Looks like there is nothing to assert, maybe just run the command itself
```suggestion
    system bin/"countdown", "0m0s"
```

Comment (antonmedv) [reply Formula/countdown.rb:15]: This is actually running the command itself. And tests parsing. 0s to make the test run faster.

Comment (chenrui333) [reply Formula/countdown.rb:15]: yeah, I understand the `0m0s` part, but I just dont understand the response assertion part. While the command has finished, why do we care about the `""` response?

Comment (chenrui333) [reply Formula/countdown.rb:15]: if `system bin/"countdown", "0m0s"` failed, the `test` step would also failed.

Comment (antonmedv) [reply Formula/countdown.rb:15]: Oh, make sense. Will drop it.

Comment (chenrui333) [reply Formula/countdown.rb]: unfortunately, we might have to switch to use your old test (it is due to the interactive nature of the command) 🤦

Comment (chenrui333) [reply Formula/countdown.rb]: Thanks for the try though.

Comment (antonmedv) [reply Formula/countdown.rb]: Tried with system and pty: both not working =( Will use pipe_output back.

**Comment by antonmedv** [comment]
I think, I fixed everything.

**Review by chenrui333 — DISMISSED** [review]

File: Formula/countdown.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+  end
+
+  test do
+    pipe_output bin/"countdown", "0m0s"
```
Comment (carlocab) [inline Formula/countdown.rb:15]:
```suggestion
    system bin/"countdown", "0m0s"
```
You're not doing anything with the output (or even piping anything to `countdown`), so we don't need `pipe_output` here.

File: Formula/countdown.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class Countdown < Formula
+  desc "Terminla countdown timer"
```
Comment (carlocab) [inline Formula/countdown.rb:2]:
```suggestion
  desc "Terminal countdown timer"
```

Comment (chenrui333) [reply Formula/countdown.rb:2]: That is a good catch! 😆

File: Formula/countdown.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/countdown.rb:15]: yeah, but we ran into some test failure as below

```
==> /opt/homebrew/Cellar/countdown/1.0.0/bin/countdown 0m0s panic: termbox: error while reading terminfo data: EOF goroutine 1 [running]: main.main() github.com/antonmedv/countdown/main.go:78 +0x2e4 Error: countdown: failed An exception occurred within a child process: BuildError: Failed executing: /opt/homebrew/Cellar/countdown/1.0.0/bin/countdown 0m0s
```

**Comment by antonmedv** [comment]
So what else should I do?

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@antonmedv, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**